src/load.ts

```diff
@@ -288,7 +291,18 @@ export default async function load(
     }
   }
 
-  // 4. scan for discriminators (after everything’s resolved in one file)
+  // 4. collect parameters (which must be hoisted to the top)
+  for (const k of Object.keys(options.schemas)) {
```

Comment:
The fix for this was adding an entire additional pass over all subschemas. This PR does make `openapi-typescript` noticably slower (don’t worry—we’re still only talking milliseconds), however, it’s still faster than `5.x` by an order of magnitude (not needing Prettier to run).

Reply:
But slow + correct is better than fast + broken.